addons/survey/static/src/js/survey_form.js

```diff
@@ -234,18 +240,14 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         }
     },
 
-    _onMatrixBtnClick: function (event) {
+    _onSimpleChoiceRadioClick: function(event) {
```

Comment:
Hello @smottet 

I have experimented a bit with this locally and I'm not a big fan of the fact that when you un-select a matrix radio, it goes through both "_onChangeChoiceItem" AND "_onSimpleChoiceRadioClick".

Could everything be handled in "_onChangeChoiceItem" so we can get rid of "_onSimpleChoiceRadioClick" ?
Looks feasible to me but maybe I'm missing something.

Please try it out and tell me how it goes.

Thanks!

Reply:
_onChangeChoiceItem is called on change. When you click on an already selected radio button (to unselect it), you don't trigger a change event. Then I would have to change the _onChangeChoiceItem trigger to an onClick event. I will look if it can work.